src/transformStateWithClones.js

```diff
@@ -7,7 +7,42 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let res = [];
+  const newState = { ...state };
+
+  res[0] = newState;
+
+  for (let i = 0; i < actions.length; i++) {
+    switch (actions[i].type) {
+      case 'addProperties':
+        const addRes = Object.assign(res[i], actions[i].extraData);
+
+        res[i + 1] = { ...addRes };
```

Comment:
Try to change other cases in similar way
```suggestion
        newState = Object.assign(newState, action.extraData);

        previousVersions.push({ ...newState });
```